### DISCUSSION TITLE
2.0.2 release

### DISCUSSION THREAD

**Discussion by eschutho** [apache/superset#22630] [General]
Forum for tracking bugs/issues/fixes for 2.0.2 release. 

Please leave any bugs or fixes that you know of for 2.0.2. As of 1/6/2023 we haven't started building the release yet, but we want to start tracking any bugs in a centralized area so that we can start to prioritize them. 

[list of all bugs that are tagged 2.0.2](https://github.com/apache/superset/issues?q=label%3Av2.0.2+is%3Aclosed): 

Bugs reported:
https://github.com/apache/superset/issues/21984
https://github.com/apache/superset/issues/22705
https://github.com/apache/superset/issues/22691
https://github.com/apache/superset/issues/22626
https://github.com/apache/superset/issues/22802 (change name of button to "create dataset")
https://github.com/apache/superset/issues/22843

**Comment by saghatelian** [comment]
I've just submitted this annoying little bug. Happy to help debug or test the fix.

https://github.com/apache/superset/issues/22691

**Reply by eschutho** [reply]
Thank you @saghatelian! It looks like @rusackas assigned someone to look at it.

**Comment by mdeshmu** [comment]
[22626](https://github.com/apache/superset/issues/22626)
[22640](https://github.com/apache/superset/issues/22640)

**Reply by eschutho** [reply]
Thank you @mdeshmu and thanks for the PR as well!

**Comment by rusackas** [comment]
https://github.com/apache/superset/pull/20648
https://github.com/apache/superset/pull/21779

These two are needed to get `just-handlebars-helpers` into the release and make the Handlebars plugin MUCH more useful. People are reporting the lack of these on 2.0.1 as a bug. It's _kind of_ a feature, but it's pretty necessary to make the plugin actually useful. First PR adds the plugin, second PR resolves a webpack issue with how the dependencies are loaded.

(both are labeled as 2.0.2)

**Reply by eschutho** [reply]
Great, thank you!

**Comment by rusackas** [comment]
https://github.com/apache/superset/pull/21157/ (which seemingly fixes https://github.com/apache/superset/issues/22705 according to @eric-briscoe)

**Comment by Scomcn** [comment]
#22208 from @Stanhoucke 

For existing dashboards, modifications imported via /api/v1/assets/import/ are only appended if the modified dashboard contains a new chart. Modifications to dashboards that remove a chart are not implemented when imported through the assets API.

This change updates the assets API import method to remove all existing references to chart ids for the imported dashboard(s) before inserting the new chart ids. The change should benefit the management of Superset assets under source control.

**Reply by eschutho** [reply]
Thank you, added!

**Comment by eschutho** [comment]
Note to pin WTforms and cryptography as per https://github.com/apache/superset/issues/22571

**Comment by zhaorui2022** [comment]
Hi. We encountered a regression in 2.0.0 and the issue [#20679](https://github.com/apache/superset/issues/20679) was fixed by cherry picking [this commit](https://github.com/apache/superset/pull/20680)

**Reply by eschutho** [reply]
Thank you. Added. 👍

**Comment by frabenetti** [comment]
Hi, just wondering if it would be possible to add the fix to the issues #22640 and #22393 in 2.0.2. Thanks in advance and kind regards.

**Reply by eschutho** [reply]
Thank you for the PR, @frabenetti! I just left a comment to see if we can get the PR merged for you.

**Reply by frabenetti** [reply]
Thanks a lot!

**Comment by rusackas** [comment]
https://github.com/apache/superset/pull/20833 will fix #22784

**Reply by eschutho** [reply]
got it, thanks!

**Comment by rusackas** [comment]
Another PR: https://github.com/apache/superset/pull/21066
Requested on this thread: https://github.com/apache/superset/pull/22921

**Reply by eschutho** [reply]
added, thanks!

**Comment by kakoni** [comment]
How about [22006](https://github.com/apache/superset/pull/22006).
Excel export has been requested for long time.

**Reply by eschutho** [reply]
Because that is a new feature, we'll see this PR go out in either 2.1 or 3.0. 2.0.2 will only include bug fixes.

**Reply by kakoni** [reply]
Ah yes, my bad. Thanks for the clarification.

**Comment by mdeshmu** [comment]
Can we add PR [22934?](https://github.com/apache/superset/pull/22934)

**Reply by eschutho** [reply]
Thanks, yes, added!

**Comment by rusackas** [comment]
One more community-requested PR: https://github.com/apache/superset/pull/21993

**Reply by eschutho** [reply]
added!

**Comment by eschutho** [comment]
We're going to start to package up this release next week, so please add any more PRs here by February 17th.

**Comment by mdeshmu** [comment]
Will it be possible to get this fixed and merged?

#17319

**Reply by eschutho** [reply]
added!

**Comment by cbuffevant** [comment]
can we include this one https://github.com/apache/superset/pull/22280?

**Reply by eschutho** [reply]
added!

**Comment by tooptoop4** [comment]
please include https://github.com/apache/superset/pull/21645

**Reply by eschutho** [reply]
added!

**Comment by padbk** [comment]
There have been some nice fixes recently, if these could be added please. Apologies if there are any in here that are fixes for features not in 2.0.0.
https://github.com/apache/superset/pull/22329
https://github.com/apache/superset/pull/22383
https://github.com/apache/superset/pull/22443
https://github.com/apache/superset/pull/22614
https://github.com/apache/superset/pull/22727
https://github.com/apache/superset/pull/22898
https://github.com/apache/superset/pull/22920
https://github.com/apache/superset/pull/22929
https://github.com/apache/superset/pull/22984
https://github.com/apache/superset/pull/22990
https://github.com/apache/superset/pull/22991
https://github.com/apache/superset/pull/23012
https://github.com/apache/superset/pull/23039

**Reply by eschutho** [reply]
thanks, that was a good list. Added. We'll see what merges in cleanly.

**Comment by mdeshmu** [comment]
Can we add?

https://github.com/apache/superset/pull/21998
https://github.com/apache/superset/pull/22770

**Reply by lilykuang** [reply]
added

**Comment by fullergalway** [comment]
It would be nice to see [22843](https://github.com/apache/superset/issues/22843) included to make life better for plugin developers.

**Comment by D3nn3** [comment]
There is an open PR for handling the exorbitant loading times on the welcome page which is already open for over a month: https://github.com/apache/superset/pull/21873

Any chance we could this done in the near future? The welcome page unfortunately is fairly useless as it currently is because of this.

**Reply by rusackas** [reply]
There are changes requested on that PR that haven't been addressed. Hopefully we can merge it soon, but I'm not sure it'll get in under the wire.

**Comment by cccs-tom** [comment]
I know I'm pushing the bugfix vs feature line, but is there any chance that #19890 could be squeezed in?

**Reply by eschutho** [reply]
Yeah, this one came up in the 2.0.1 release as well, and it was pretty risky. We had to follow up with quite a bit more bug fixes. I'd suggest leaving this one until 2.1.0

**Reply by cccs-tom** [reply]
Understood - looking forward to it! And thanks for checking! :)

**Comment by mdeshmu** [comment]
Please include https://github.com/apache/superset/pull/22321

**Reply by eschutho** [reply]
added!

**Comment by mdeshmu** [comment]
Please include

https://github.com/apache/superset/pull/20613
https://github.com/apache/superset/pull/23139

**Comment by eschutho** [comment]
Thank you to everyone who commented here! We took at pass at cherrying in the PRs for 2.0.2, but found that less than a third of them actually merged in cleanly, even after some time spent trying to fix conflicts and update tests. Since we're already close to an RC2 on 2.1.0 it seems that there would be more value in concentrating our efforts on that release. It doesn't seem that 2.0.2 will have a lot of the bug fixes that people are looking for because of the divergence of code at this point, but they'll all be in 2.1.0, so we're going to cancel the 2.0.2 release. If anyone has any feedback, though, we'd love to hear it.

**Reply by mdeshmu** [reply]
Does that mean 2.0.2 will never be released and support for 2.0.x series will end here?

**Reply by rusackas** [reply]
Probably, but not _for sure_. It's most likely that 2.1+ releases will be the ones receiving patches from here out, though sometimes things like security patches get pulled further back. It's all a volunteer effort building/testing these releases, so efforts tend to go toward newer and (in theory, at least) more widely adopted releases.
